src/relax/transform/to_mixed_precision.cc

```diff
@@ -0,0 +1,297 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*!
+ * \file src/relax/transform/to_mixed_precision.cc
+ * \brief Automatic mixed precision pass.
+ */
+
+#include <tvm/relax/expr_functor.h>
+#include <tvm/relax/op_attr_types.h>
+#include <tvm/relax/transform.h>
+
+#include "../op/make_op.h"
+
+namespace tvm {
+namespace relax {
+
+enum MixedTypePolicy : int {
+  MIXED_PRECISION_ALWAYS = 0,
+  MIXED_PRECISION_FOLLOW = 1,
+  MIXED_PRECISION_NEVER = 2
+};
+
+// Return array is of type : [MixedTypePolicy (int), bool, Call]
+// Call is a call node, out_dtype_str is the expected output_dtype string
+using FMixedPrecision = runtime::TypedPackedFunc<Array<ObjectRef>(
+    const Call& call_node, const std::string& out_dtype_str)>;
```

Comment:
Use DataType instead of string of out_dtype. Python side's string will be automatically converted to DataType

Reply:
Does it work reversely, which means C side's DataType will be automatically converted to Python side's string?

Reply:
Yes

Reply:
I tried this, it reports the following error.
```
pyargs = (C_TO_PY_ARG_SWITCH[type_codes[i]](args[i]) for i in range(num_args))

KeyError: 5'
```

Reply:
are you passing it as return value or argument? Perhaps the argument conversion is not as perfect atm

Reply:
Try to register in C++ for now given this is mostly trivial

Reply:
I'm passing it as argument.